### PR TITLE
[15.0][IMP] product_cost_security: Add a security group to modify the cost price.

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT36548

ping @chienandalu @CarlosRoca13 

With this PR the stock_account_product_cost_security module is not needed anymore.

### CODE REVIEW

File: product_cost_security/README.rst [hunk 1]
```diff
@@ -25,8 +25,7 @@ Product Cost Security
 
 |badge1| |badge2| |badge3| |badge4| |badge5| 
 
-Adds a security group for product cost price field.
-Only users with this group can view this field.
+Adds two security groups, one for view product cost price and other for editing it
```
Comment (pedrobaeza) [inline product_cost_security/README.rst]:
```suggestion
It adds two security groups, one for viewing the product cost price, and the other for editing it.
```

File: product_cost_security/__manifest__.py [hunk 2]
```diff
@@ -3,7 +3,7 @@
 {
     "name": "Product Cost Security",
     "summary": "Product cost security restriction view",
-    "version": "15.0.1.0.0",
+    "version": "15.0.1.0.1",
```
Comment (pedrobaeza) [inline product_cost_security/__manifest__.py:6]: Why manually changing the version? Does this require migration script when coming from v14? If the `id` of the XML-ID has been preserved, you don't need to do anything.

File: product_cost_security/models/product_product.py [hunk 3]
```diff
@@ -7,3 +7,6 @@ class ProductProduct(models.Model):
     _inherit = "product.product"
 
     standard_price = fields.Float(groups="product_cost_security.group_product_cost")
+    user_can_update_cost = fields.Boolean(
```
Comment (pedrobaeza) [inline product_cost_security/models/product_product.py]: This is not needed, as inheritance by delegation does exactly this.

File: product_cost_security/models/product_template.py [hunk 4]
```diff
@@ -7,3 +7,12 @@ class ProductTemplate(models.Model):
     _inherit = "product.template"
 
     standard_price = fields.Float(groups="product_cost_security.group_product_cost")
+    user_can_update_cost = fields.Boolean(compute="_compute_user_can_update_cost")
+
+    def _compute_user_can_update_cost(self):
```
Comment (pedrobaeza) [inline product_cost_security/models/product_template.py:13]: This compute should depend on the user:

```suggestion
    @api.depends_context("uid")
    def _compute_user_can_update_cost(self):
```

**Comment by pedrobaeza** [comment]
pre-commit CI fixed in #1187

**Comment by sergio-teruel** [comment]
@pedrobaeza Changes done!!

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
:+1:

**Pull request merged**